cloud_sql/postgres_instance_psa_psc/main.tf

```
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START cloud_sql_postgres_instance_psa_psc]
+
+# [START vpc_postgres_instance_private_ip_network]
+resource "google_compute_network" "peering_network" {
+  name                    = "private-network"
+  auto_create_subnetworks = "false"
+}
+# [END vpc_postgres_instance_private_ip_network]
+
+# [START vpc_postgres_instance_private_ip_address]
+resource "google_compute_global_address" "private_ip_address" {
+  name          = "private-ip-address"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = google_compute_network.peering_network.id
+}
+# [END vpc_postgres_instance_private_ip_address]
+
+# [START vpc_postgres_instance_private_ip_service_connection]
+resource "google_service_networking_connection" "default" {
+  network                 = google_compute_network.peering_network.id
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.private_ip_address.name]
+}
+# [END vpc_postgres_instance_private_ip_service_connection]
+
+# [START cloud_sql_postgres_instance_psa_psc_instance]
+resource "google_sql_database_instance" "default" {
+  name             = "postgres-instance"
+  region           = "us-central1"
+  database_version = "POSTGRES_14"
+
+  depends_on = [google_service_networking_connection.default]
+
+  settings {
+    tier              = "db-custom-2-7680"
+    availability_type = "REGIONAL"
+    backup_configuration {
+      enabled = true
+    }
+    ip_configuration {
+      psc_config {
+        psc_enabled               = true
+        allowed_consumer_projects = [] # Add consumer project IDs here.
+      }
+      ipv4_enabled    = false
+      private_network = google_compute_network.peering_network.id
+    }
+  }
+  deletion_protection = false # Set to "true" to prevent destruction of the resource
```

For the MySQL equivalent file we have a more detailed comment for `deletion_protection`, can we add it here to remain consistent?

```suggestion
  # set `deletion_protection` to true, will ensure that one cannot accidentally delete this instance by
  # use of Terraform whereas `deletion_protection_enabled` flag protects this instance at the GCP level.
  deletion_protection = false
```